examples/vision/sam.py

```diff
@@ -0,0 +1,599 @@
+"""
+Title: Segment Anything Model with 🤗Transformers
+Authors: [Merve Noyan](https://twitter.com/mervenoyann) & [Sayak Paul](https://twitter.com/RisingSayak)
+Date created: 2023/07/11
+Last modified: 2023/07/11
+Description: Implementing Segment Anything Model using 🤗Transformers.
+Accelerator: GPU
+"""
+"""
+## Introduction
+
+Large language models (LLMs) make it easy for the end users to apply them to various
+applications through "prompting". For example if we wanted an LLM to predict the
+sentiment of the following sentence -- "That movie was amazing, I thoroughly enjoyed it"
+-- we'd do prompt the LLM with something like:
+
+> What's the sentiment of the following sentence: "That movie was amazing, I thoroughly
+enjoyed it"?
+
+In return, the LLM would return sentiment token.
+
+But when it comes to visual recognition tasks, how can we engineer "visual" cues to
+prompt foundation vision models? For example, we could have an input image and prompt the
+model with bounding box on that image and ask it to perform segmentation. The bounding
+box would serve as our visual prompt here.
+
+In the [Segment Anything Model](https://segment-anything.com/) (dubbed as SAM),
+researchers from Meta extended the space of language prompting to visual prompting. SAM
+is capable of performing zero-shot segmentation with a prompt input, inspired by large
+language models. The prompt here can be a set of foreground/background points, free text,
+a box or a mask. There are many downstream segmentation tasks, including semantic
+segmentation and edge detection. The goal of SAM is to enable all of these downstream
+segmentation tasks through prompting.
+
+In this example, we'll learn how to use the SAM model from 🤗 Transformers for performing
+inference and fine-tuning.
+
+
+"""
+
+"""
+## Installation
+"""
+
+"""shell
+!pip install -q git+https://github.com/huggingface/transformers
+"""
+
+"""
+## SAM in a few words
+
+SAM has the following components:
+
+|
+![](https://imgur.com/oLfdwuB)
+|:--:|
+| Image taken from the official [SAM blog
+post](https://ai.facebook.com/blog/segment-anything-foundation-model-image-segmentation/)
```

Comment:
```suggestion
| Image taken from the official [SAM blog post](https://ai.facebook.com/blog/segment-anything-foundation-model-image-segmentation/) |
```

I have seen that links should be in a single line.